news/unsafe-rust-in-the-wild-notes-on-the-current-state-of-unsafe-rust.md

```diff
@@ -0,0 +1,72 @@
+  stake, software foundations, technology consortiums, and governments around
+  the world are taking notice and advocating for improved programming practices
+  and tools. 
+date: 2024-05-21T12:30:00Z
+tags:
+  - rust
+  - foundation
+  - security
+index: true
+layout: layouts/news.njk
+---
+Programs access memory – this concept is fundamental to computing. Depending on the programming language used to write code, there are various ways to manage the allocation and use of that memory. Each approach requires care and caution.
+
+Unintended, out-of-scope access to a program’s memory regions can expose sensitive data or serve as an entry point for exploitation, posing significant risk and potentially contributing to zero-day attacks. In short, whether a programmer manages and allocates memory manually or relies on the language and compiler to do it for them, robust memory management practices are a necessity.
+
+Memory safety issues account for a significant share of software vulnerabilities. \[1\] Malicious actors are well aware of this and use an evolving set of tactics to exploit memory-unsafe code in some of the most recognizable and damaging software attacks in recent years, such as Heartbleed.
+
+With so many damaging software exploits haunting our industry and so much at stake, software foundations, technology consortiums, and governments around the world are taking notice and advocating for improved programming practices and tools.
+
+## The Power and Promise of Rust
+
+The Rust programming language is frequently cited by memory-safety advocates as one of the most secure programming languages on the market today. A memory-safe language by default through a concept called ownership, Rust provides rules and guarantees for memory management and centers security as a first-class concept.
+
+Programs using Rust are *unable* to compile if memory management rules are violated, essentially eliminating the possibility of a memory issue at runtime. This provides Rust developers and users of applications written in Rust with a high degree of confidence that memory vulnerabilities need not be a concern.
+
+## Safe Rust and “Unsafe Rust”
+
+Although Rust is a powerful tool for memory-safety and security, the power of its safety checks can become limiting when the program cannot actually go wrong but the compiler is unable to guarantee that itself; the programmer can prove the impossibility of undefined behavior in ways not available to the compiler. In these instances, Rust programmers will apply the \`unsafe\` keyword to indicate a function or code segment which has a) additional safety considerations or b) invariants a programmer must manually follow to guarantee safety that are not necessarily expressible by Rust or the function itself. The \`unsafe\` keyword enables developers to dereference a raw pointer, modify a mutable static variable, and, crucially, call unsafe functions.
```

Comment:
Backticks will work! Just looked at an example of another blog of ours that had them so we're good. Feel free to use them throughout.

Reply:
the backticks will work, but not if they are escaped with the backslash in front. with `` \` `` the blog will only show \`.